### PR TITLE
add "variables" argument to `argspec.kwonlyaards`

### PR DESCRIPTION
**Pull request by drezap**
**no unit tests written**
**not tested locally**

This is related to these two pull issues:

https://github.com/tensorflow/tensorflow/issues/31945
https://github.com/tensorflow/tensorflow/issues/38622

What this should do is force the input of `grad()` function of class `Loss` to forget an extra argument required when instantiating a `Variable()` with the custom loss function.

Ideally I'd implement the patch and build from source, but I'm encountering build from source issues via bazel: `https://github.com/tensorflow/tensorflow/issues/38736`, likely due to Fedora.

Let me know what need be changed.

It'd be awesome to be able to build from source so that I can test the patch, but bazel is still messing up, if one of the devs knows a quick fix I can implement it.

### CODE REVIEW

**Comment by drezap** [comment]
@googlebot I signed it!

**Review by alextp — APPROVED** [review]

**Comment by gbaned** [comment]
@drezap  Can you please address Ubuntu Sanity errors? Thanks!

**Comment by drezap** [comment]
yes easy fix not now too busy. soon

**Comment by gbaned** [comment]
@drezap Any update on this PR? Please. Thanks!

**Comment by FranzKnut** [comment]
This bug remains unsolved then?
